src/global/pidfile.cc

```diff
@@ -96,3 +96,27 @@ int pidfile_remove(void)
   pid_file[0] = '\0';
   return 0;
 }
+
+int test_pid_file_in_use(const md_config_t *conf) 
+{
+  int fd;
+  if (conf->pid_file.empty()) {
+    return pidfile_remove();
```

Comment:
it's better to exit rather than return 0. if return 0,  daemon will run without pid_file, which acts against this patch.